### DISCUSSION TITLE
使用Python的numpy第三方函式庫(third party library)=程式包(套件，package)

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#182] [General]
我想使用Visual Studio Code，在numpy_test.py的「腳本(script)」中，建立一個shape為(2,3)的二維陣列(array)。
# 第一步：出現NameError : name 'numpy' is not defined。代表目前使用的Python直譯器沒有找到numpy。
![image](https://user-images.githubusercontent.com/86505454/144692670-0ed7a554-4b27-42f2-abdf-63ed0535c3f8.png)
# 第二步：我使用Visual Studio Code的終端機執行numpy_test.py，但什麼都沒有跑出來
<img width="960" alt="numpy跑不出東西" src="https://user-images.githubusercontent.com/86505454/144693061-3e2bc6c7-2d75-4bbc-9ff0-1525fdfec7e0.png">

**Comment by Jie2222** [comment]
第一步看不到因為沒有先import numpy，所以第二步如果print(x)因該就會看到正確的結果了

**Reply by enoch-lightning** [reply]
1.我在程式後加上print(x)，果然就看到x輸出到電腦瑩幕上。是合理的，因為print(x)的功能就是將x輸出到電腦螢幕上。
![image](https://user-images.githubusercontent.com/86505454/145148701-0b33afac-3b8e-48c6-8c6e-27f92bc02c50.png)

**Reply by enoch-lightning** [reply]
2.若此Python直譯器，無法依照相對路徑(relative path)找到numpy(寫成Numpy也可以)，便會在Visual Studio Code的「終端機(terminal)」顯示Error。Error的中文就是「錯誤。」

**Reply by enoch-lightning** [reply]
3.此時，只要下載並安裝numpy，就可以使用import匯入numpy這個模組(module)。

**Reply by enoch-lightning** [reply]
Python的import與java的import、C與C++的#include是一樣的意思。

**Comment by chia-kai-chang** [comment]
可以試試：
## 方法1
```
import numpy as np
x=np.array([[1,2,3],[4,5,6]])
x
```
![image](https://user-images.githubusercontent.com/1768891/145198086-34838870-d1b6-4827-ab80-be35f8f10a3d.png)


## 方法2
```
import numpy as np
x=np.array([1,2,3,4,5,6])
x.reshape(2,3)
```
![image](https://user-images.githubusercontent.com/1768891/145198012-32b61480-f376-4660-90f2-bca1886a4b2c.png)


## 方法3
```
import numpy as np
x=np.arange(6)
x.reshape(2,3)
```
![image](https://user-images.githubusercontent.com/1768891/145197959-1c050cdf-96db-476b-8ad9-a00739d999d3.png)

**Reply by enoch-lightning** [reply]
好的，謝謝老師。這三種方式應該都是建立一個shape為(2,3)的「陣列(array)」，不是矩陣(matrix)。
